networkit/cpp/graph/GraphBuilder.cpp

```diff
@@ -213,24 +226,26 @@ void GraphBuilder::toGraphParallel(Graph &G) {
                                         // undirected graphs
                 inEdgesPerThread[tid][u].push_back(v);
                 if (weighted) {
-                    edgeweight ew = outEdgeWeights[v][i];
-                    inWeightsPerThread[tid][u].push_back(ew);
+                    G.addPartialEdge(unsafe, v, u, outEdgeWeights[v][i]);
+                    inWeightsPerThread[tid][u].push_back(outEdgeWeights[v][i]);
+                } else {
+                    G.addPartialEdge(unsafe, v, u);
                 }
```

Comment:
Is adding edges safe here? I assume no two threads access the same `v` at the same time?

Reply:
yes it is safe